test/e2e/helpers.js

```diff
@@ -147,9 +147,46 @@ async function withFixtures(options, testSuite) {
   }
 }
 
+const getWindowHandles = async (driver) => {
+  await driver.waitUntilXWindowHandles(3);
```

Comment:
What's the significance of 3 window handles? Why not 2 or 4?

Reply:
When using test dapp there are 3 windows:
1. main extension window where user metamask login is done
2. text dapp window
3. extension popup

Wait above ensure that all windows are opened.

Reply:
What if the dapp is not started, but this is called? The name of this function sounds like it can be used to get the window handles for _any_ kind of test, regardless of whether it involves the test dapp or not.

Reply:
It might make sense to add a code comment as to what each of the windows represents, if only for maintainability.

Reply:
I'm wondering if it would help to change the name of this function to reflect its intended use. Maybe something like `getWindowHandlesIncludingDappServer`? Just thinking that if the dapp server is not started, then `waitUntilXWindowHandles(3)` will literally wait for a window to exist that will never appear.

Reply:
I just updated the PR, thanks for the useful feedbacks 👍